### ISSUE TITLE
Local Media ordering is wrong for albums larger than 100 tracks

### ISSUE THREAD

**Issue by aereaux** [ncmpcpp/ncmpcpp#196]
In the Local Media view, it seems to be sorting the tracks lexicographically by track number.  Is there some way to fix this so it sorts them numerically?

**Comment by arybczak** [comment]
Do you mean media library screen? It should sort them as numbers there. Can you show an example?

**Comment by aereaux** [comment]
Sorry, it's in the browser screen.  Here's part of one of my albums that has over 100 tracks:
```
[..]                                                                                                                                                                                  
Jean Sibelius - Symphony no. 1 in E minor, op. 39: I. Andante, ma non troppo - Allegro energico                                                                                 (9:50)
Jean Sibelius - Symphony no. 1 in E minor, op. 39: II. Andante, ma non troppo lento                                                                                             (8:58)
Jean Sibelius - Symphony no. 1 in E minor, op. 39: III. Scherzo. Allegro                                                                                                        (4:31)
Jean Sibelius - Symphony no. 1 in E minor, op. 39: IV. Finale (Quasi una Fantasia). Andante - Allegro molto                                                                    (12:09)
Jean Sibelius - Symphony no. 4 in A minor, op. 63: I. Tempo molto moderato, quasi adagio                                                                                       (11:40)
Jean Sibelius - Symphony no. 4 in A minor, op. 63: II. Allegro molto vivace                                                                                                     (4:38)
Jean Sibelius - Symphony no. 4 in A minor, op. 63: III. Il tempo largo                                                                                                         (14:06)
Jean Sibelius - Symphony no. 4 in A minor, op. 63: IV. Allegro                                                                                                                  (9:41)
Jean Sibelius - Symphony no. 2 in D major, op. 43: I. Allegretto                                                                                                                (9:24)
Jean Sibelius - Symphony no. 2 in D major, op. 43: II. Tempo Andante, ma rubato                                                                                                (14:30)
Jean Sibelius - Soi kiitokseski luojan, op. 23 no. 6a                                                                                                                           (1:37)
Jean Sibelius - Män från slätten och havet, op. 65a                                                                                                                             (4:48)
Jean Sibelius - Till havs, op. 84, No. 5                                                                                                                                        (2:29)
Jean Sibelius - Fridolins dårskap, JS 84                                                                                                                                        (2:34)
Jean Sibelius - Jone havsfärd, JS 100                                                                                                                                           (2:37)
Jean Sibelius - Jääkärien marssi, op. 91a                                                                                                                                       (2:30)
Jean Sibelius - Finlandia-hymni, op. 26                                                                                                                                         (2:35)
Jean Sibelius - Svarta rosor, op. 36 no. 1                                                                                                                                      (2:01)
Jean Sibelius - Men min fågel märks dock icke, op. 36 no. 2                                                                                                                     (2:30)
Jean Sibelius - Säv, säv, susa, op. 36 no. 4                                                                                                                                    (2:40)
Jean Sibelius - Symphony no. 2 in D major, op. 43: III. Vivacissimo                                                                                                             (5:58)
Jean Sibelius - Illalle, op. 17 no. 6                                                                                                                                           (1:22)
Jean Sibelius - Im feld ein mädchen singt, op. 50 no. 3                                                                                                                         (3:04)
Jean Sibelius - Aus banger brust, op. 50 no. 4                                                                                                                                  (2:25)
Jean Sibelius - Die stille stadt, op. 50 no. 5                                                                                                                                  (2:47)
Jean Sibelius - Våren flyktar hastigt, op. 13 no. 4                                                                                                                             (1:37)
Jean Sibelius - Se'n har jag ej frågat mera, op. 17 no. 1                                                                                                                       (2:01)
Jean Sibelius - Höstkväll, op. 38 no. 1                                                                                                                                         (4:36)
Jean Sibelius - Arioso, op. 3                                                                                                                                                   (3:57)
Jean Sibelius - Näcken, JS 138                                                                                                                                                  (3:28)
Jean Sibelius - Giv mig ej glans, ej guld, ej prakt, op. 1 no. 4                                                                                                                (3:50)
Jean Sibelius - Symphony no. 2 in D major, op. 43: IV. Finale. Allegro moderato                                                                                                (15:17)
Jean Sibelius - Jag är ett träd, op. 57 no. 5                                                                                                                                   (2:54)
Jean Sibelius - Näcken, op. 57 no. 8                                                                                                                                            (2:16)
Jean Sibelius - Hymn to Thaïs, JS 97                                                                                                                                            (2:01)
Jean Sibelius - Kaiutar, op. 72 no. 4                                                                                                                                           (3:17)
Jean Sibelius - Längtan heter min arvedel, op. 86 no. 2                                                                                                                         (3:17)
Jean Sibelius - I systrar, I bröder, I älskande par, op. 86 no. 6                                                                                                               (2:11)
Jean Sibelius - Demanten på marssnön, op. 36 no. 6                                                                                                                              (3:04)
Jean Sibelius - Sången om korsspindeln, op. 27 no. 4                                                                                                                            (3:38)
Jean Sibelius - På verandan vid havet, op. 38 no. 2                                                                                                                             (3:14)
Jean Sibelius - Kom nu hit, död, op. 60 no. 1                                                                                                                                   (2:56)
Jean Sibelius - Symphony no. 3 in C major, op. 52: I. Allegro moderato                                                                                                         (10:24)
Jean Sibelius - Souda, souda, sinisorsa, JS 180                                                                                                                                 (1:16)
Jean Sibelius - Norden, op. 90 no. 1                                                                                                                                            (2:11)
Jean Sibelius - Narciss, JS 140                                                                                                                                                 (1:51)
Jean Sibelius - Den första kyssen, op. 37 no. 1                                                                                                                                 (2:00)
Jean Sibelius - Var det en dröm?, op. 37 no. 4                                                                                                                                  (2:05)
Jean Sibelius - Flickan kom ifrån sin älsklings möte, op. 37 no. 5                                                                                                              (2:57)
Jean Sibelius - Tanken, JS 192                                                                                                                                                  (2:01)
Jean Sibelius - Moderato - Allegro Appassionato in C-sharp minor, JS 131                                                                                                       (10:46)
Jean Sibelius - String Quartet in A minor, JS 183: I. Andante - Allegro                                                                                                        (11:14)
Jean Sibelius - String Quartet in A minor, JS 183: II. Adagio ma non tanto                                                                                                      (8:38)
Jean Sibelius - Symphony no. 3 in C major, op. 52: II. Andantino con moto, quasi allegretto                                                                                    (11:18)
Jean Sibelius - String Quartet in A minor, JS 183: III. Vivace                                                                                                                  (5:47)
Jean Sibelius - String Quartet in A minor, JS 183: IV. Allegro                                                                                                                  (9:00)
Jean Sibelius - String Quartet in D minor "Voces intimae", op. 56: I. Andante - Allegro molto moderato                                                                          (5:54)
Jean Sibelius - String Quartet in D minor "Voces intimae", op. 56: II. Vivace                                                                                                   (2:21)
Jean Sibelius - String Quartet in D minor "Voces intimae", op. 56: III. Adagio di molto                                                                                        (10:59)
Jean Sibelius - String Quartet in D minor "Voces intimae", op. 56: IV. Allegretto, ma pesante                                                                                   (5:37)
Jean Sibelius - String Quartet in D minor "Voces intimae", op. 56: V. Allegro                                                                                                   (5:27)
Jean Sibelius - Piano Trio in D Major, "Korpo", JS 209: I. Allegro moderato                                                                                                    (10:57)
Jean Sibelius - Piano trio in, D major, "Korpo", JS 209: II. Fantasia. Andante - molto adagio - andantino - moderato - andantino                                               (15:31)
Jean Sibelius - Piano Trio in D major, "Korpo", JS 209: III. Finale. Vivace                                                                                                     (9:38)
Jean Sibelius - Symphony no. 3 in C major, op. 52: III. Moderato - Allegro (ma non tanto)                                                                                       (9:22)
Jean Sibelius - Piano quintet in G minor, JS 159: I. Grave - Allegro                                                                                                           (10:59)
Jean Sibelius - Piano quintet in G minor, JS 159: II. Intermezzo, moderato                                                                                                      (5:25)
Jean Sibelius - Piano quintet in G minor, JS 159: III. Andante                                                                                                                  (9:24)
Jean Sibelius - Piano quintet in G minor, JS 159: IV. Scherzo, vivacissimo                                                                                                      (3:23)
Jean Sibelius - Piano quintet in G minor, JS 159: V. Moderato - Vivace                                                                                                         (10:01)
Jean Sibelius - La pompeuse marche d'asis, JS 116                                                                                                                               (3:51)
Jean Sibelius - Vattendroppar, JS 216                                                                                                                                           (0:55)
Jean Sibelius - Andante Cantabile in E-flat major, JS 30b                                                                                                                       (4:23)
Jean Sibelius - Andante Cantabile in G major, JS 33                                                                                                                             (2:40)
Jean Sibelius - Sonata in F major, JS 178: I. Allegro                                                                                                                          (11:18)
Jean Sibelius - Symphony no. 5 in E-flat major, op. 82: I. Tempo molto moderato, allegro moderato                                                                              (13:30)
Jean Sibelius - Sonata in F major, JS 178: II. Andante                                                                                                                          (6:03)
Jean Sibelius - Sonata in F major, JS 178: III. Vivace                                                                                                                          (8:07)
Jean Sibelius - Malinconia, op. 20                                                                                                                                             (12:32)
Jean Sibelius - Four Pieces for Violin or Cello and Piano, op. 78 no. 1: Impromptu                                                                                              (1:46)
Jean Sibelius - Four Pieces for Violin or Cello and Piano, op. 78 no. 2: Romance in F major                                                                                     (3:28)
Jean Sibelius - Four Pieces for Violin or Cello and Piano, op. 78 no. 3: Religioso                                                                                              (4:07)
Jean Sibelius - Four Pieces for Violin or Cello and Piano, op. 78 no. 4: Rigaudon                                                                                               (2:08)
Jean Sibelius - Sonatina for Violin and Piano in E major, op. 80: I. Lento-Allegro                                                                                              (3:55)
Jean Sibelius - Sonatina for Violin and Piano in E major, op. 80: II. Andantino                                                                                                 (3:58)
Jean Sibelius - Sonatina for Violin and Piano in E major, op. 80: III. Lento-Allegretto                                                                                         (4:12)
Jean Sibelius - Symphony no. 5 in E-flat major, op. 82: II. Andante mosso, quasi allegretto                                                                                     (8:47)
Jean Sibelius - Two Danses Champêtres for Violin and Piano from op. 106: I. Largamente assai. Vivace                                                                            (5:08)
Jean Sibelius - Two Danses Champêtres for Violin and Piano from op. 106: II. Alla polacca                                                                                       (2:57)
Jean Sibelius - Florestan, JS 82: I. Moderato                                                                                                                                   (1:38)
Jean Sibelius - Florestan, JS 82: II. Molto moderato                                                                                                                            (2:24)
Jean Sibelius - Florestan, JS 82: III. Andante                                                                                                                                  (4:08)
```
I'm not sure how to get track numbers to show there, so I added them (in the order they appear there) to my playlist:
```
Jean Sibelius                       01/19 Symphony no. 1 in E minor, op. 39: I. Andante, ma non troppo - Allegro energico            The Essential Sibelius                    9:50
Jean Sibelius                       02/19 Symphony no. 1 in E minor, op. 39: II. Andante, ma non troppo lento                        The Essential Sibelius                    8:58
Jean Sibelius                       03/19 Symphony no. 1 in E minor, op. 39: III. Scherzo. Allegro                                   The Essential Sibelius                    4:31
Jean Sibelius                       04/19 Symphony no. 1 in E minor, op. 39: IV. Finale (Quasi una Fantasia). Andante - Allegro molt The Essential Sibelius                    12:09
Jean Sibelius                       05/19 Symphony no. 4 in A minor, op. 63: I. Tempo molto moderato, quasi adagio                   The Essential Sibelius                    11:40
Jean Sibelius                       06/19 Symphony no. 4 in A minor, op. 63: II. Allegro molto vivace                                The Essential Sibelius                    4:38
Jean Sibelius                       07/19 Symphony no. 4 in A minor, op. 63: III. Il tempo largo                                     The Essential Sibelius                    14:06
Jean Sibelius                       08/19 Symphony no. 4 in A minor, op. 63: IV. Allegro                                             The Essential Sibelius                    9:41
Jean Sibelius                       09/19 Symphony no. 2 in D major, op. 43: I. Allegretto                                           The Essential Sibelius                    9:24
Jean Sibelius                       10/19 Symphony no. 2 in D major, op. 43: II. Tempo Andante, ma rubato                            The Essential Sibelius                    14:30
Jean Sibelius                       100/1 Soi kiitokseski luojan, op. 23 no. 6a                                                      The Essential Sibelius                    1:37
Jean Sibelius                       101/1 Män från slätten och havet, op. 65a                                                        The Essential Sibelius                    4:48
Jean Sibelius                       102/1 Till havs, op. 84, No. 5                                                                   The Essential Sibelius                    2:29
Jean Sibelius                       103/1 Fridolins dårskap, JS 84                                                                   The Essential Sibelius                    2:34
Jean Sibelius                       104/1 Jone havsfärd, JS 100                                                                      The Essential Sibelius                    2:37
Jean Sibelius                       105/1 Jääkärien marssi, op. 91a                                                                  The Essential Sibelius                    2:30
Jean Sibelius                       106/1 Finlandia-hymni, op. 26                                                                    The Essential Sibelius                    2:35
Jean Sibelius                       107/1 Svarta rosor, op. 36 no. 1                                                                 The Essential Sibelius                    2:01
Jean Sibelius                       108/1 Men min fågel märks dock icke, op. 36 no. 2                                                The Essential Sibelius                    2:30
Jean Sibelius                       109/1 Säv, säv, susa, op. 36 no. 4                                                               The Essential Sibelius                    2:40
Jean Sibelius                       11/19 Symphony no. 2 in D major, op. 43: III. Vivacissimo                                        The Essential Sibelius                    5:58
Jean Sibelius                       110/1 Illalle, op. 17 no. 6                                                                      The Essential Sibelius                    1:22
Jean Sibelius                       111/1 Im feld ein mädchen singt, op. 50 no. 3                                                    The Essential Sibelius                    3:04
Jean Sibelius                       112/1 Aus banger brust, op. 50 no. 4                                                             The Essential Sibelius                    2:25
Jean Sibelius                       113/1 Die stille stadt, op. 50 no. 5                                                             The Essential Sibelius                    2:47
Jean Sibelius                       114/1 Våren flyktar hastigt, op. 13 no. 4                                                        The Essential Sibelius                    1:37
Jean Sibelius                       115/1 Se'n har jag ej frågat mera, op. 17 no. 1                                                  The Essential Sibelius                    2:01
Jean Sibelius                       116/1 Höstkväll, op. 38 no. 1                                                                    The Essential Sibelius                    4:36
Jean Sibelius                       117/1 Arioso, op. 3                                                                              The Essential Sibelius                    3:57
Jean Sibelius                       118/1 Näcken, JS 138                                                                             The Essential Sibelius                    3:28
Jean Sibelius                       119/1 Giv mig ej glans, ej guld, ej prakt, op. 1 no. 4                                           The Essential Sibelius                    3:50
Jean Sibelius                       12/19 Symphony no. 2 in D major, op. 43: IV. Finale. Allegro moderato                            The Essential Sibelius                    15:17
Jean Sibelius                       120/1 Jag är ett träd, op. 57 no. 5                                                              The Essential Sibelius                    2:54
Jean Sibelius                       121/1 Näcken, op. 57 no. 8                                                                       The Essential Sibelius                    2:16
Jean Sibelius                       122/1 Hymn to Thaïs, JS 97                                                                       The Essential Sibelius                    2:01
Jean Sibelius                       123/1 Kaiutar, op. 72 no. 4                                                                      The Essential Sibelius                    3:17
Jean Sibelius                       124/1 Längtan heter min arvedel, op. 86 no. 2                                                    The Essential Sibelius                    3:17
Jean Sibelius                       125/1 I systrar, I bröder, I älskande par, op. 86 no. 6                                          The Essential Sibelius                    2:11
Jean Sibelius                       126/1 Demanten på marssnön, op. 36 no. 6                                                         The Essential Sibelius                    3:04
Jean Sibelius                       127/1 Sången om korsspindeln, op. 27 no. 4                                                       The Essential Sibelius                    3:38
Jean Sibelius                       128/1 På verandan vid havet, op. 38 no. 2                                                        The Essential Sibelius                    3:14
Jean Sibelius                       129/1 Kom nu hit, död, op. 60 no. 1                                                              The Essential Sibelius                    2:56
Jean Sibelius                       13/19 Symphony no. 3 in C major, op. 52: I. Allegro moderato                                     The Essential Sibelius                    10:24
Jean Sibelius                       130/1 Souda, souda, sinisorsa, JS 180                                                            The Essential Sibelius                    1:16
Jean Sibelius                       131/1 Norden, op. 90 no. 1                                                                       The Essential Sibelius                    2:11
Jean Sibelius                       132/1 Narciss, JS 140                                                                            The Essential Sibelius                    1:51
Jean Sibelius                       133/1 Den första kyssen, op. 37 no. 1                                                            The Essential Sibelius                    2:00
Jean Sibelius                       134/1 Var det en dröm?, op. 37 no. 4                                                             The Essential Sibelius                    2:05
Jean Sibelius                       135/1 Flickan kom ifrån sin älsklings möte, op. 37 no. 5                                         The Essential Sibelius                    2:57
Jean Sibelius                       136/1 Tanken, JS 192                                                                             The Essential Sibelius                    2:01
Jean Sibelius                       137/1 Moderato - Allegro Appassionato in C-sharp minor, JS 131                                   The Essential Sibelius                    10:46
Jean Sibelius                       138/1 String Quartet in A minor, JS 183: I. Andante - Allegro                                    The Essential Sibelius                    11:14
Jean Sibelius                       139/1 String Quartet in A minor, JS 183: II. Adagio ma non tanto                                 The Essential Sibelius                    8:38
Jean Sibelius                       14/19 Symphony no. 3 in C major, op. 52: II. Andantino con moto, quasi allegretto                The Essential Sibelius                    11:18
Jean Sibelius                       140/1 String Quartet in A minor, JS 183: III. Vivace                                             The Essential Sibelius                    5:47
Jean Sibelius                       141/1 String Quartet in A minor, JS 183: IV. Allegro                                             The Essential Sibelius                    9:00
Jean Sibelius                       142/1 String Quartet in D minor "Voces intimae", op. 56: I. Andante - Allegro molto moderato     The Essential Sibelius                    5:54
Jean Sibelius                       143/1 String Quartet in D minor "Voces intimae", op. 56: II. Vivace                              The Essential Sibelius                    2:21
Jean Sibelius                       144/1 String Quartet in D minor "Voces intimae", op. 56: III. Adagio di molto                    The Essential Sibelius                    10:59
Jean Sibelius                       145/1 String Quartet in D minor "Voces intimae", op. 56: IV. Allegretto, ma pesante              The Essential Sibelius                    5:37
Jean Sibelius                       146/1 String Quartet in D minor "Voces intimae", op. 56: V. Allegro                              The Essential Sibelius                    5:27
Jean Sibelius                       147/1 Piano Trio in D Major, "Korpo", JS 209: I. Allegro moderato                                The Essential Sibelius                    10:57
Jean Sibelius                       148/1 Piano trio in, D major, "Korpo", JS 209: II. Fantasia. Andante - molto adagio - andantino  The Essential Sibelius                    15:31
Jean Sibelius                       149/1 Piano Trio in D major, "Korpo", JS 209: III. Finale. Vivace                                The Essential Sibelius                    9:38
Jean Sibelius                       15/19 Symphony no. 3 in C major, op. 52: III. Moderato - Allegro (ma non tanto)                  The Essential Sibelius                    9:22
Jean Sibelius                       150/1 Piano quintet in G minor, JS 159: I. Grave - Allegro                                       The Essential Sibelius                    10:59
Jean Sibelius                       151/1 Piano quintet in G minor, JS 159: II. Intermezzo, moderato                                 The Essential Sibelius                    5:25
Jean Sibelius                       152/1 Piano quintet in G minor, JS 159: III. Andante                                             The Essential Sibelius                    9:24
Jean Sibelius                       153/1 Piano quintet in G minor, JS 159: IV. Scherzo, vivacissimo                                 The Essential Sibelius                    3:23
Jean Sibelius                       154/1 Piano quintet in G minor, JS 159: V. Moderato - Vivace                                     The Essential Sibelius                    10:01
Jean Sibelius                       155/1 La pompeuse marche d'asis, JS 116                                                          The Essential Sibelius                    3:51
Jean Sibelius                       156/1 Vattendroppar, JS 216                                                                      The Essential Sibelius                    0:55
Jean Sibelius                       157/1 Andante Cantabile in E-flat major, JS 30b                                                  The Essential Sibelius                    4:23
Jean Sibelius                       158/1 Andante Cantabile in G major, JS 33                                                        The Essential Sibelius                    2:40
Jean Sibelius                       159/1 Sonata in F major, JS 178: I. Allegro                                                      The Essential Sibelius                    11:18
Jean Sibelius                       16/19 Symphony no. 5 in E-flat major, op. 82: I. Tempo molto moderato, allegro moderato          The Essential Sibelius                    13:30
Jean Sibelius                       160/1 Sonata in F major, JS 178: II. Andante                                                     The Essential Sibelius                    6:03
Jean Sibelius                       161/1 Sonata in F major, JS 178: III. Vivace                                                     The Essential Sibelius                    8:07
Jean Sibelius                       162/1 Malinconia, op. 20                                                                         The Essential Sibelius                    12:32
Jean Sibelius                       163/1 Four Pieces for Violin or Cello and Piano, op. 78 no. 1: Impromptu                         The Essential Sibelius                    1:46
Jean Sibelius                       164/1 Four Pieces for Violin or Cello and Piano, op. 78 no. 2: Romance in F major                The Essential Sibelius                    3:28
Jean Sibelius                       165/1 Four Pieces for Violin or Cello and Piano, op. 78 no. 3: Religioso                         The Essential Sibelius                    4:07
Jean Sibelius                       166/1 Four Pieces for Violin or Cello and Piano, op. 78 no. 4: Rigaudon                          The Essential Sibelius                    2:08
Jean Sibelius                       167/1 Sonatina for Violin and Piano in E major, op. 80: I. Lento-Allegro                         The Essential Sibelius                    3:55
Jean Sibelius                       168/1 Sonatina for Violin and Piano in E major, op. 80: II. Andantino                            The Essential Sibelius                    3:58
Jean Sibelius                       169/1 Sonatina for Violin and Piano in E major, op. 80: III. Lento-Allegretto                    The Essential Sibelius                    4:12
Jean Sibelius                       17/19 Symphony no. 5 in E-flat major, op. 82: II. Andante mosso, quasi allegretto                The Essential Sibelius                    8:47
Jean Sibelius                       170/1 Two Danses Champêtres for Violin and Piano from op. 106: I. Largamente assai. Vivace       The Essential Sibelius                    5:08
Jean Sibelius                       171/1 Two Danses Champêtres for Violin and Piano from op. 106: II. Alla polacca                  The Essential Sibelius                    2:57
Jean Sibelius                       172/1 Florestan, JS 82: I. Moderato                                                              The Essential Sibelius                    1:38
Jean Sibelius                       173/1 Florestan, JS 82: II. Molto moderato                                                       The Essential Sibelius                    2:24
```
As you can see, it goes from 01-10 to 100-109 to 11, and so on.

I'm using ncmpcpp from commit c3b996c in case it matters.

**Comment by ZotanWolf** [comment]
I am having the exact same issue, hope this can be resolved

**Comment by arybczak** [comment]
Fixed by 6f8d12da8ce6a7c7a7e1494c773f7219bddb9049.
